runtime/runtime-params-estimator/src/utils.rs

```diff
@@ -233,26 +237,47 @@ pub(crate) fn fn_cost_in_contract(
         testbed.process_block(vec![setup_tx], 0);
     }
 
-    let blocks = {
-        let mut blocks = Vec::with_capacity(n_blocks);
-        for account in chosen_accounts {
-            let tb = testbed.transaction_builder();
-            let mut block = Vec::new();
-            for _ in 0..block_size {
-                let tx = tb.transaction_from_function_call(account.clone(), method, Vec::new());
-                block.push(tx);
-            }
-            blocks.push(block);
-        }
-        blocks
-    };
+    let mut blocks = Vec::with_capacity(n_blocks);
+    // Measurement blocks with single tx with many actions.
+    for account in chosen_accounts.drain(..n_blocks) {
+        let actions = std::iter::repeat_with(|| function_call_action(method.to_string()))
+            .take(n_actions)
+            .collect();
+        let tx = testbed.transaction_builder().transaction_from_actions(
+            account.clone(),
+            account,
+            actions,
+        );
+        blocks.push(vec![tx]);
+    }
+    // Base with single tx with single action. Insert it after warm-up blocks.
+    let final_account = chosen_accounts.pop().unwrap();
+    let base_tx = testbed.transaction_builder().transaction_from_actions(
+        final_account.clone(),
+        final_account,
+        vec![function_call_action(method.to_string())],
+    );
+    blocks.insert(n_warmup_blocks, vec![base_tx]);
 
     let mut measurements = testbed.measure_blocks(blocks, 0);
-    measurements.drain(0..ctx.config.warmup_iters_per_block);
+    measurements.drain(0..n_warmup_blocks);
 
+    let (base_gas_cost, _base_ext_costs) = measurements.remove(0);
+    // Do not subtract block overhead because we already subtract the base.
+    let overhead = None;
+    let block_size = 1;
     let (gas_cost, _ext_costs) =
-        aggregate_per_block_measurements(ctx.config, block_size, measurements, Some(overhead));
-    gas_cost
+        aggregate_per_block_measurements(ctx.config, block_size, measurements, overhead);
+    gas_cost.saturating_sub(&base_gas_cost, &NonNegativeTolerance::Strict) / (n_actions - 1) as u64
+}
+
+fn function_call_action(method_name: String) -> Action {
+    Action::FunctionCall(near_primitives::transaction::FunctionCallAction {
```

Comment:
```suggestion
    Action::FunctionCall(FunctionCallAction {
```

I think actions are intended to be imported

Reply:
done